src/message/__tests__/messageActionSheet-test.js

```diff
@@ -56,60 +56,98 @@ describe('constructActionButtons', () => {
   });
 });
 
-describe('constructHeaderActionButtons', () => {
+describe('constructTopicActionButtons', () => {
+  test('show mark as read if topic is unread', () => {
+    const stream = eg.makeStream({ name: 'test stream', stream_id: 123 });
+    const subscriptions = [{ ...eg.subscription, ...stream }];
+    const unreadStreams = eg.makeUnreadStreamsState([
+      { streamId: 123, topics: [{ topicName: 'test topic', messageIds: [1, 2] }] },
+    ]);
+    const buttons = constructTopicActionButtons({
+      backgroundData: { ...baseBackgroundData, subscriptions, unreadStreams },
+      streamId: stream.stream_id,
+      topic: 'test topic',
```

Comment:
In a related spirit, instead of saying the string `'test topic'` twice and implicitly using that they're the same, it'd be better to have a `const topic = 'test topic'` and refer to that in both places.